### PR TITLE
Add simple config schema resource

### PR DESCRIPTION
**Pull request by dawehner**
Supports https://github.com/jsdrupal/drupal-admin-ui/pull/70

### CODE REVIEW

File: admin_ui_support/src/Plugin/rest/resource/ConfigSchemaResource.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Provides a rest resource for config schema.
+ *
+ * @RestResource(
+ *   id = "config_schema",
+ *   label = @Translation("Config schema"),
+ *   uri_paths = {
+ *     "canonical" = "/admin-api/config-schema/{name}"
```
Comment (dawehner) [inline admin_ui_support/src/Plugin/rest/resource/ConfigSchemaResource.php]: Note: For now I went with one resource plugin without any derivative. I'm not sure there is any crucial information in config schema not safe to export.

File: admin_ui_support/tests/src/Functional/rest/ConfigSchemaResourceTest.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    $this->refreshTestStateAfterRestConfigChange();
+  }
+
+  public function testConfigSchema() {
+    $this->initAuthentication();
+    $url = Url::fromRoute('rest.config_schema.GET', ['_format' => static::$format, 'name' => 'system.site']);
+    $request_options = $this->getAuthenticationRequestOptions('GET');
+
+    $response = $this->request('GET', $url, $request_options);
+    $this->assertResourceResponse(200, false, $response, ['config:rest.resource.config_schema', 'config:rest.settings', 'http_response'], [], FALSE, 'MISS');
```
Comment (tedbow) [inline admin_ui_support/tests/src/Functional/rest/ConfigSchemaResourceTest.php]: I get `Failed asserting that 403 is identical to 200.` here

Comment (dawehner) [reply admin_ui_support/tests/src/Functional/rest/ConfigSchemaResourceTest.php]: Oh, mh, I actually just c&pasted the test together :(

Comment (tedbow) [reply admin_ui_support/tests/src/Functional/rest/ConfigSchemaResourceTest.php]: This fixes the test https://github.com/tedbow/drupal-admin-ui/commit/03ea0a8339fd2a27230feb63376562cea6437708

Comment (dawehner) [reply admin_ui_support/tests/src/Functional/rest/ConfigSchemaResourceTest.php]: Thank you!

**Comment by dawehner** [comment]
Suggestion: We get this in to be able to iterate. This will require serious work over time anyway.

**Comment by dawehner** [comment]
This will be replaced by https://github.com/jsdrupal/drupal-admin-ui/pull/97 over time.

**Pull request closed without merging**